src/core/templatetags/ascii_encode.py

```diff
@@ -0,0 +1,8 @@
+from django import template
+
+register = template.Library()
+
+@register.filter
+def ascii_encode(string):
+    """ ASCII-encode a string, replacing non-ASCII chars with HTML elements. """
+    return string.encode('ascii', 'xmlcharrefreplace')
```

Comment:
You're doing this over and over throughout this process. Wouldn't it be useful to create a single method that does this, limiting the possibility of getting `.encode('ascii', 'xmlcharrefreplace')` wrong with any of the points it's used?